src/profile_manager/views.py

```diff
@@ -149,6 +156,71 @@ def dispatch(self, *args, **kwargs):
         else:
             raise Http404("Sorry, this profile isn't yours!")
 
+    def post(self, request, *args, **kwargs):
+        forms = self.get_forms()
+
+        # check if all form instances are valid else ...
+        if all([form.is_valid() for form in forms]):
+            return self.form_valid(forms)
+        return self.form_invalid(forms)
+
+    def get_context_data(self, **kwargs):
+        profile = self.get_object()
+        context = {}
+
+        # return instance of form or new form instance
+        context['forms'] = kwargs.get('forms', self.get_forms())
+
+        context['heading'] = "Editing " + profile.user.get_username() + "'s Profile"
+        context['submit_btn_value'] = "Update"
+        context['profile'] = profile
+
+        return context
+
+    # handles form_kwargs for all forms  # redundant?
+    def get_form_kwargs(self, initial={}):
+        kwargs = initial
+
+        # if request is POST send request data+files
+        if self.request.method == "POST":
+            kwargs.update({
+                "data": self.request.POST,
+                "files": self.request.FILES,
+            })
+
+        return kwargs
+
+    # returns instance of ProfileForm
+    def get_form_public(self):
+        return self.public_form_class(**self.get_form_kwargs(
+            initial={'instance': self.get_object()}
+        ))
+
+    # returns instance of UserForm
+    def get_form_staff(self):
+        return self.staff_form_class(**self.get_form_kwargs(
+            initial={'instance': self.get_object().user}
+        ))
+
+    # runs if all forms are valid
+    def form_valid(self, forms, *args, **kwargs):
+        # log back in since you get kicked out if you change your own username
+        if self.request.user == self.get_object().user:
```

Comment:
Instead of logging the user back in, can we redirect to the log in page (does it do that by default?)  Although,  I tried commenting this code out and then changing my username and it didn't kick me out anyway?

Reply:
hmm before when userform was on its own page it would log me out